### DISCUSSION TITLE
Why so pervasive, how can I get it off my machine?

### DISCUSSION THREAD

**Discussion by boardtc** [gitbutlerapp/gitbutler#5584] [General]
I have to get this off my machine. It's dreadful how it took over my commit messages and workspace.
But there is no running process in Task Manager to kill to allow the uninstall to proceed.
I opened a bug https://github.com/gitbutlerapp/gitbutler/issues/5583

**Comment by boardtc** [comment]
I was able to remove after a reboot but when I open IntelliJ my terminal window still has (gitbutler/workspace) written after it. How can I get rid of this?

**Reply by Byron** [reply]
Thanks for opening the issue. Unfortunately there was no way to do it without the workspace branch.

While GitButler is installed, it should clear itself out of a project by removing it from the project list.

<img width="689" alt="Screenshot 2024-11-18 at 13 41 07" src="https://github.com/user-attachments/assets/2c609fd9-4de2-4664-8096-49a867ba7afb">

The screenshot above was taken from the project settings.

Something that should be easier is to remove all projects - the following screenshot was taken after clicking the 'account' button on the bottom-left.

<img width="985" alt="Screenshot 2024-11-18 at 13 41 59" src="https://github.com/user-attachments/assets/836ec644-0461-4e19-878d-8a0dd88fb310">

However, I am not sure it will actually reset each of these repositories to a pre-GitButler state.

If GitButler was already removed, one can remove the per-project files by hand.

First, let's get back to the `main` branch: `git checkout main`. Now local metadata can be removed with `rm -Rf .git/gitbutler`. Finally, one can remove GitButler branches with `rm -Rf .git/refs/gitbutler/ .git/refs/heads/gitbutler/` .

Project-independent application state as [described here](https://docs.gitbutler.com/development/debugging#logs) would certainly have been removed as part of the uninstaller, but I didn't validate it.

I hope that helps.

**Comment by boardtc** [comment]
Thanks for the prompt reply. 

I also found \.git\logs\refs\heads\gitbutler, 
.config:
[log]
	excludeDecoration = refs/gitbutler	
It also changed the jenkins author on my build I loaded to gitbutler :-(

**Reply by Byron** [reply]
Right, there is more indeed, thanks for sharing it here for completeness.

Something I don't really understand is how the crediting happened - GitButler showing up in your commits means this *opt-in* checkbox is enabled.

<img width="664" alt="Screenshot 2024-11-18 at 16 25 17" src="https://github.com/user-attachments/assets/a627d707-b2d2-4156-9f96-6af26c6757ad">

Was this enabled for you by default, or did it get enabled by accident?

**Comment by tacurran** [comment]
@boardtc @Byron I ran into the same uninstall issue, with the added problem that for whatever reason the MACOS GITBUTLER app does not start. Even after reinstalling and restarting the machine. So take a look at the #14491

**Reply by Byron** [reply]
It should definitely be able to start no matter what - can you try `/Applications/GitButler.app/Contents/MacOS/gitbutler-tauri` from a terminal and see if there is something in the logs? Thank you.
